src/App.tsx

```diff
@@ -1,59 +1,66 @@
-import React from 'react';
+import React, { useContext } from 'react';
+import cn from 'classnames';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
-import classNames from 'classnames';
+import { UserSelector } from './components/UserSelector';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
-import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
+import { UsersContext } from './store/UsersContext';
+import { PostsContext } from './store/PostsContext';
+
 export const App: React.FC = () => {
+  const { selectedUser } = useContext(UsersContext);
+  const { posts, loading, errorMessage, selectedPost } =
+    useContext(PostsContext);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
-              <div className="block">
-                <UserSelector />
-              </div>
-
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">No user selected</p>
-
-                <Loader />
-
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
-
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
+                <div className="block">
+                  <UserSelector />
                 </div>
 
-                <PostsList />
+                {loading ? (
```

Comment:
nested ternary is not the easiest thing to read

instead, make your conditions more specific, but separated:
`{loading && !errorMessage && <Loader />}`

It also might not look really good if the condition contains a lot of variables, so you can create separate variables above, like:
`const showLoader = loading and !errorMessage;`
and so on

Reply:
Same in PostDetails